src/generic/graphicc.cpp

```diff
@@ -2749,6 +2762,31 @@ void wxCairoContext::Flush()
 #endif
 }
 
+void wxCairoContext::GetDPI(wxDouble* dpiX, wxDouble* dpiY) const
+{
+    if ( GetWindow() )
+    {
+        const wxSize dpi = GetWindow()->GetDPI();
+
+        if ( dpiX )
+            *dpiX = dpi.x;
+        if ( dpiY )
+            *dpiY = dpi.y;
+    }
+    else
+    {
+        double dpi = (double)wxDisplay::GetStdPPIValue();
+#ifndef __WXMAC__
+        dpi *= GetContentScaleFactor();
+#endif
```

Comment:
Again, this seems to indicate that there is something different with Cairo under Mac, but what?